src/tests/tracing/eventcounter/gh53564.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if USE_MDT_EVENTSOURCE
+using Microsoft.Diagnostics.Tracing;
+#else
+using System.Diagnostics.Tracing;
+#endif
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+namespace gh53564Tests
+{
+    public class RuntimeCounterListener : EventListener
+    {
+        public RuntimeCounterListener(){}
+
+        private DateTime? setToZeroTimestamp = null;
+        private DateTime? mostRecentTimestamp = null;
+        public ManualResetEvent ReadyToVerify { get; } = new ManualResetEvent(initialState: false);
+
+        protected override void OnEventSourceCreated(EventSource source)
+        {
+            if (source.Name.Equals("System.Runtime"))
+            {
+                Dictionary<string, string> refreshInterval = new Dictionary<string, string>();
+
+                Console.WriteLine($"[{DateTime.Now:hh:mm:ss.fff}] Setting interval to 1");
+                // first set interval to 1 seconds
+                refreshInterval["EventCounterIntervalSec"] = "1";
+                EnableEvents(source, EventLevel.Informational, (EventKeywords)(-1), refreshInterval);
+
+                // wait a moment to get some events
+                Thread.Sleep(TimeSpan.FromSeconds(3));
+
+                // then set interval to 0
+                Console.WriteLine($"[{DateTime.Now:hh:mm:ss.fff}] Setting interval to 0");
+                refreshInterval["EventCounterIntervalSec"] = "0";
```

Comment:
```suggestion
                refreshInterval["EventCounterIntervalSec"] = "0.00001";
```
I suspect this slightly more evil version of the test still hangs? It will no longer go into the `if(pollingIntervalInSeconds <= 0)` branch but due to the floating point math it is still going to set `_pollingIntervalInMilliseconds = 0`

I'd suggest rather than trying to identify every possible flow of code and data that can get that loop to spin infinitely eliminate the possibility by making it a for loop with a limited number of possible iterations. We'd never expect it to go more than a couple iterations so if for some reason it looped 1000 times then something is clearly not running properly and we should just give up.
The other issue is that every additional line changed outside that do/while loop raises the question "given that this change will happen even when the app is not in the infinite loop situation, what if this is introducing a new bug somewhere else?" If we can avoid making changes to any other lines its less code we have to carefully analyze and defend the correctness of.